src/shims/unix/solarish/foreign_items.rs

```diff
@@ -82,6 +82,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let result = this.macos_fbsd_solaris_readdir_r(dirp, entry, result)?;
                 this.write_scalar(result, dest)?;
             }
+            // TODO readdir ?
```

Comment:
Please leave readdir for a future PR.

Reply:
I have to because test fs do not pass, it is just a reminder.

Reply:
You can just disable parts of the test on Solarish for now. Please keep the PR small.